### PR TITLE
chore(sdk): wrap conditional branch groups in outer pipeline

### PR DESCRIPTION
**Pull request by connor-mccarthy**
**Description of your changes:**
This change is a pre-requisite for collecting outputs from conditional branches. Each conditional branch (if, elif, else) must be wrapped in a common outer pipeline which can surface "one of" the inner task outputs via [`value_from_oneof`](https://github.com/kubeflow/pipelines/blob/45fe8e86583646143d1685d9e04d887ff27440ee/api/v2alpha1/pipeline_spec.proto#L165) for parameters and [`artifact_selectors`](https://github.com/kubeflow/pipelines/blob/45fe8e86583646143d1685d9e04d887ff27440ee/api/v2alpha1/pipeline_spec.proto#L124) for artifacts.

No functional change. There is a UI change.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more 
about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/test_data/pipelines/if_elif_else_complex.py [hunk 1]
```diff
@@ -77,7 +82,7 @@ def lucky_number_pipeline(add_drumroll: bool = True,
                         text='Announcing again: Got the lucky number 5000! A one in 10,000 chance.'
                     )
 
-    print_strings(strings=dsl.Collected(even_or_odd_task.output))
+    print_ints(ints=dsl.Collected(int_task.output))
```
Comment (connor-mccarthy) [inline sdk/python/test_data/pipelines/if_elif_else_complex.py:85]: The previous state was erroneous and the expected behavior is ambiguous. Updated this example and created a work item to add validation for this case.

**Review by chensun — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**